File: madameautomata-bot-python-autoreg/bot/commands/worship.py

import discord
from discord import app_commands, Interaction
from ..theme import gothic_embed
from ..util.perm import is_domme, is_staff, is_sub
from ..db import Session, User, Collaring, Cage, JailSession, Star, Task, Worship, Config, CollarType, StarStatus
from ..util.time import parse_duration, random_case
from ..util.ids import cuid
from datetime import datetime, timedelta
from sqlalchemy import select, func, update, and_

def setup(bot):
    @bot.tree.command(name="worship", description="Offer worship to the Worship Me role or a Domme.")
    @app_commands.describe(domme="Domme to worship", text="Your worship text")
    async def worship(i: Interaction, text: str, domme: discord.User | None = None):
        async with Session() as s:
            s.add(Worship(id=cuid(), sub_id=i.user.id, domme_id=(domme.id if domme else None), content=text))
            cfg = await s.get(Config, 1)
            await s.commit()
        e = gothic_embed("Worship", f"**<@{i.user.id}>** whispers:\n> {text}\n\n{f'For <@{domme.id}>' if domme else ''}")
        if cfg and cfg.worship_channel_id:
            ch = i.guild.get_channel(int(cfg.worship_channel_id))
            if ch and hasattr(ch, "send"):
                await ch.send(embed=e)
                return await i.response.send_message("Offered.", ephemeral=True)
        await i.response.send_message(embed=e)

File: madameautomata-js-bot/src/commands/worship.js

import { SlashCommandBuilder, PermissionFlagsBits, ChannelType, ActionRowBuilder, ButtonBuilder, ButtonStyle, StringSelectMenuBuilder, EmbedBuilder } from "discord.js";
import { prisma } from "../lib/prisma.js";
import { Theme, gothicEmbed } from "../theme/theme.js";
import { parseDuration, randomCase } from "../util/duration.js";
import { cuid } from "../util/ids.js";
import { isDomme, isSub, isStaff, getConfig } from "../util/perm.js";

const cmd = 
{
  data: new SlashCommandBuilder()
    .setName("worship")
    .setDescription("Offer worship to the Worship Me role or a Domme.")
    .addUserOption(o=>o.setName("domme").setDescription("Domme to worship"))
    .addStringOption(o=>o.setName("text").setDescription("Your worship text").setRequired(true)),
  async run(i) {
    const domme = i.options.getUser("domme");
    const text = i.options.getString("text", true);
    await prisma.worship.create({ data: { id: cuid(), subId: i.user.id, dommeId: domme?.id || null, content: text } });
    const cfg = await getConfig();
    const e = gothicEmbed("Worship", `**<@${i.user.id}>** whispers:\n> ${text}\n\n${domme?`For <@${domme.id}>`:""}`);
    const ch = cfg.worshipChannelId ? i.guild.channels.cache.get(cfg.worshipChannelId) : null;
    if (ch?.isTextBased?.()) { await ch.send({ embeds: [e] }); return i.reply({ content: "Offered.", ephemeral: true }); }
    return i.reply({ embeds: [e] });
  }
}


export default cmd;
